synthetic/image/parser.py

```diff
@@ -0,0 +1,126 @@
+import json
+import logging
+import random
+from functools import partial
+from pathlib import Path
+from typing import Type
+
+from .poisson_blending import blend_image
+from .synthesizer import ImageSynthesizer
+
+from PIL import ImageFont, Image
+
+
+logger = logging.getLogger(__name__)
+
+
+class AlreadyProcessed(Exception):
+    pass
+
+
+class NoBoundingBoxes(Exception):
+    pass
+
+
+def synthesize_image(
+    image_file,
+    dst_dir,
+    font_path,
+    font_size_range,
+    jitter,
+    json_file,
+    max_size,
+    num_outputs_per_document,
+    synthesizer_class,
+):
+    def _out_path(_i, suffix):
+        return dst_dir / f'{json_file.stem}-{_i}{suffix}'
+
+    k_to_process = []
+    for i in range(num_outputs_per_document):
+        if not (_out_path(i, '.jpeg').exists() and _out_path(i, '.json').exists()):
+            k_to_process.append(i)
+
+    if not k_to_process:
+        raise AlreadyProcessed(f'Already processed {image_file} {json_file}')
+
+    ground_truth = json.loads(json_file.read_text())
+    from_size, to_size = font_size_range
+    fonts = {i: ImageFont.truetype(str(font_path), i) for i in range(to_size, from_size, -1)}
```

Comment:
why is this done in reversed order?

Reply:
We wish to try fit the biggest font size first